garak/generators/huggingface.py

```diff
@@ -50,53 +52,134 @@ def _set_hf_context_len(self, config):
             if isinstance(config.n_ctx, int):
                 self.context_len = config.n_ctx
 
+    def _gather_hf_params(self, hf_constructor: Callable):
+        # this may be a bit too naive as it will pass any parameter valid for the pipeline signature
+        # this falls over when passed `from_pretrained` methods as the callable model params are not explicit
```

Comment:
Is it worth catching this and gathering those params differently?

Reply:
This is mitigated by adding `Pipeline` params when `model` is not a parameter of the callable, future iterations may suggest adding a supported set of params so I left the notes here to offer details on `why` the code adds more parameters in some cases.